app/models/portfolio.rb

```diff
@@ -11,14 +11,20 @@
 
 
 class Portfolio < ApplicationRecord
+  include Discard::Model
   acts_as_tenant(:tenant)
+  default_scope -> { kept }
 
-  validates :name, presence: true, :uniqueness => { :scope => :tenant_id }
+  validates :name, :presence => true, :uniqueness => { :scope => %i(tenant_id discarded_at) }
   validates :image_url, :format => { :with => URI::DEFAULT_PARSER.make_regexp }, :allow_blank => true
   validates :enabled_before_type_cast, :format => { :with => /\A(true|false)\z/i }, :allow_blank => true
 
   has_many :portfolio_items, :dependent => :destroy
 
+  after_discard do
+    portfolio_items.discard_all
```

Comment:
@lindgrenj6 If we have an exception, I think we should raise it ( store it ) and continue on with the other deletions. The UI will then be able to know what was 'discarded' and what still exists.

@gmcculloug - We've discussed this before I believe.  If not please add your thoughts.

Reply:
You mean "continue with the other `discard-tions`"  😉

This brings up a good point, this should be a `before` action.  If we cannot discard all the items we should not discard the portfolio.  Otherwise the user would lose access to the item from the UI since they are only accessed through the portfolio.

I would agree that we should try to delete as many as possible so the only remaining ones are the problematic ones.

Reply:
@gmcculloug @syncrou added a potential solution to this - have a look!